src/Hpack/Config.hs

```diff
@@ -270,6 +301,252 @@ data PackageConfig = PackageConfig {
 
 instance HasFieldNames PackageConfig
 
+instance ToJSON Package where
+  toJSON p =
+      removeEmptyArrays $
+      removeEmptyObjects $
+      (\(Object o) -> Object $ case packageSourceRepository p of
+              Just repo ->
+                  let srepo = (sourceRepositoryUrl repo) ++
+                          (fromMaybe "" (sourceRepositorySubdir repo))
+                  in
+                  if githubBaseUrl `isPrefixOf` srepo
+                     then let dropIfGH (Just (String v))
+                                  | githubBaseUrl `isPrefixOf` T.unpack v = Nothing
+                              dropIfGH v = v
+                          in
+                          HashMap.insert "github" (String (T.pack (drop (length githubBaseUrl) srepo))) $
+                          HashMap.alter dropIfGH "bug-reports" $
```

Comment:
If it's me, I would try to avoid this kind of logic here and instead do things like this in a conversion step.